cpp/src/replace/replace.cu

```diff
@@ -182,51 +198,139 @@ void replace_nulls_with_column(gdf_size_type size, Type* out_data, gdf_valid_typ
  *        `replace_nulls` with the apropiate data types.
  */
 /* ----------------------------------------------------------------------------*/
-struct replace_nulls_kernel_forwarder {
+struct replace_nulls_column_kernel_forwarder {
   template <typename col_type>
-  void operator()(gdf_size_type           nrows,
-                  gdf_size_type           new_values_length,
-                  void*            d_col_data,
-                  gdf_valid_type*  d_col_valid,
-                  const void*      d_new_value)
+  void operator()(gdf_size_type    nrows,
+                  void*            d_in_data,
+                  gdf_valid_type*  d_in_valid,
+                  const void*      d_replacement,
+                  void*            d_out_data)
```

Comment:
Consider adding a cudaStream_t parameter (default to `0`) here and pass it to all kernels, RMM_ALLOC, RMM_FREE, CHECK_STREAM and replace all `cudaMemcpy` with `cudaMemcpyAsync`. Even though this will default to the null stream (synchronous) for now, it will make it easier in the future to make the public APIs asynchronous.

Reply:
Done